File: frontend/src/services/chatService.js
import axios from 'axios';

// Đảm bảo không có dấu / ở cuối URL
const API_URL = 'http://localhost:8080/api/public/chat'; 

export const sendChatMessage = async (message) => {
    const token = localStorage.getItem('token'); 
    
    // Sử dụng dấu huyền ` ` để truyền biến token chính xác
    const config = {
        headers: token ? { Authorization: `Bearer ${token}` } : {} 
    };

    try {
        const response = await axios.post(API_URL, { message }, config);
        return response.data;
    } catch (error) {
        console.error("Lỗi AI Chat:", error);
        throw error;
    }
};

File: mobile/lib/services/chat_service.dart
import 'package:dio/dio.dart';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class ChatService {
  final Dio _dio = Dio(BaseOptions(
    baseUrl: 'http://10.0.2.2:8080/api/public', // Đường dẫn API public chat
    connectTimeout: const Duration(seconds: 30),
    receiveTimeout: const Duration(seconds: 30),
  ));

  final _storage = const FlutterSecureStorage();

  Future<Options> _getAuthOptions() async {
    String? token = await _storage.read(key: 'jwt_token');
    return Options(headers: {
      if (token != null) 'Authorization': 'Bearer $token',
      'Content-Type': 'application/json',
    });
  }

  Future<String> sendMessage(String message) async {
    try {
      final options = await _getAuthOptions();
      final response = await _dio.post(
        '/chat',
        data: {'message': message},
        options: options,
      );
      // Backend trả về chuỗi String (nội dung từ Gemini)
      return response.data.toString();
    } on DioException catch (e) {
      return "Error: ${e.response?.data ?? "Connection failed"}";
    }
  }
}